backend/app/controllers/users.js

```diff
@@ -1214,16 +1214,24 @@ function* insertOne(req, res, next) {
     var isExistUser = yield * common.isExistsUserInRealm(req, req.params.realm, req.body.email);
     if (isExistUser) {
         isExistUser.registered = true;
-
         // If user is found in table we check to see if it's been marked as deleted and un-mark it
         if (isExistUser.isDeleted !== null) {
-            const updateObj = {
-                isDeleted: null
-            };
 
-            yield thunkQuery(
-                User.update(updateObj).where(User.email.equals(req.body.email))
-            );
+            // make sure user is on auth before reactivating
+            const userExistOnAuth = yield _getUserOnAuthService(req.body.email, req.headers.authorization);
```

Comment:
I was thinking of moving the `_getUserOnAuthService` or `_createUserOnAuthService` sections into their own function because they were called like 3 times. Now they're called 6. I think this whole "check if exists, then pluck authId, else make user, pluck authId" routine should be its own function.

Reply:
Yeah, I agree. I created a ticket for it- to clean up repeated code
https://jira.amida-tech.com/browse/INBA-624